sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorClient/MetricFeedbackLiveTests.cs

```diff
@@ -0,0 +1,290 @@
+
+            Assert.That(addedFeedback.Type, Is.EqualTo(FeedbackType.Anomaly));
+
+            var anomalyFeedback = addedFeedback as MetricAnomalyFeedback;
+
+            Assert.That(anomalyFeedback, Is.Not.Null);
+            Assert.That(anomalyFeedback.AnomalyValue, Is.EqualTo(AnomalyValue.AutoDetect));
+            Assert.That(anomalyFeedback.StartTime, Is.EqualTo(CreatedFeedbackStartTime));
+            Assert.That(anomalyFeedback.EndTime, Is.EqualTo(CreatedFeedbackEndTime));
+
+            if (populateOptionalMembers)
+            {
+                Assert.That(anomalyFeedback.AnomalyDetectionConfigurationId, Is.EqualTo(DetectionConfigurationId));
+                // TODO: Add snapshot validation (https://github.com/azure/azure-sdk-for-net/issues/15915).
+            }
+            else
+            {
+                Assert.That(anomalyFeedback.AnomalyDetectionConfigurationId, Is.Null);
+                Assert.That(anomalyFeedback.AnomalyDetectionConfigurationSnapshot, Is.Null);
+            }
+        }
+
+        [RecordedTest]
+        public async Task AddAndGetChangePointFeedback()
+        {
+            MetricsAdvisorClient client = GetMetricsAdvisorClient();
+
+            var dimensionKey = new DimensionKey();
+            dimensionKey.AddDimensionColumn("city", "Delhi");
+            dimensionKey.AddDimensionColumn("category", "Handmade");
+
+            var filter = new FeedbackDimensionFilter(dimensionKey);
+
+            var feedbackToAdd = new MetricChangePointFeedback(MetricId, filter, CreatedFeedbackStartTime, CreatedFeedbackEndTime, ChangePointValue.AutoDetect);
+
+            string feedbackId = await client.AddFeedbackAsync(feedbackToAdd);
+
+            Assert.That(feedbackId, Is.Not.Null);
+
+            MetricFeedback addedFeedback = await client.GetFeedbackAsync(feedbackId);
+
+            ValidateMetricFeedback(addedFeedback, feedbackId, dimensionKey);
+
+            Assert.That(addedFeedback.Type, Is.EqualTo(FeedbackType.ChangePoint));
+
+            var changePointFeedback = addedFeedback as MetricChangePointFeedback;
+
+            Assert.That(changePointFeedback, Is.Not.Null);
+            Assert.That(changePointFeedback.ChangePointValue, Is.EqualTo(ChangePointValue.AutoDetect));
+        }
+
+        [RecordedTest]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task AddAndGetCommentFeedback(bool populateOptionalMembers)
+        {
+            MetricsAdvisorClient client = GetMetricsAdvisorClient();
+
+            var dimensionKey = new DimensionKey();
+            dimensionKey.AddDimensionColumn("city", "Delhi");
+            dimensionKey.AddDimensionColumn("category", "Handmade");
+
+            var filter = new FeedbackDimensionFilter(dimensionKey);
+            var comment = "Feedback created in a .NET test.";
+
+            var feedbackToAdd = new MetricCommentFeedback(MetricId, filter, comment);
+
+            if (populateOptionalMembers)
```

Comment:
My intent with this test case is not really testing the time range functionality, but making sure we can parse the request/response for all members. When `populateOptionalMembers = true`, we're making sure values are being passed and returned correctly. When `populateOptionalMembers = false`, we're making sure we can still make a request with the minimum configuration and we can parse the responses with null/empty/default values.

This is a pattern I've been following in MA. Let me know if you have any concerns!

Reply:
Hmm I wonder if it will be better to separate this then into different methods. One with minimum, and one with all option. This way the confusion I had about testing functionality is not assumed and the intention of the test is clear

Reply:
Setting and checking the optional properties are usually a very small portion of the test, so we'll have lots of repeated code if we separate them into two methods.

> My intent with this test case is not really testing the time range functionality, but making sure we can parse the request/response for all members. When populateOptionalMembers = true, we're making sure values are being passed and returned correctly. When populateOptionalMembers = false, we're making sure we can still make a request with the minimum configuration and we can parse the responses with null/empty/default values.

What do you think about adding docstrings to these tests (with `<summary>` tags) explaining the reasoning above?

Reply:
Yeah that could work. If in the future it turns out confusing or hard to keep track, then it can be split.